### PR TITLE
Preserve line endings in source to tidy

### PR DESCRIPTION
**Pull request by kenneth-olwing**
Use the *_raw variants of path->slurp/spew/append in all places
in order to ensure to preserve both incoming and outgoing line
endings on all platforms.

Use case when experiencing: A project developed on both Windows and Linux uses PerlTidy, specifically with the option '-ole=unix', i.e. no matter what input line endings are or on what platform we are, output the tidied source using unix lineendings. It was noticed that no matter how the flag was set, on Windows we always got CRLF line endings. Diagnosed to come from various reading/writing activities in CT, but on Windows the Perl read/writes typically always convert incoming/outgoing LF's to CRLF. By consistently using 'binmode', this will not happen. On Unix/Linux this is a no-op (but will do the right thing even if CRLF is encountered).

Not terribly proud of the added tests, as I admit to be being a bit confused by how to insert new tests, but at least they pass on both Windows/Linux :-)

### CODE REVIEW

**Comment by kenneth-olwing** [comment]
FYI, I'm trying to remedy problems with untidy code etc...BRB

**Comment by autarch** [comment]
Don't worry about the perltidy side of the tests too much. I can always fix that locally. But it might be that you have a different version of `Perl::Tidy` installed then what CI ends up with (CI should always use the latest version).

**Comment by autarch** [comment]
I just pushed a commit to master to tidy all the code with the latest Perl::Tidy. Some of the failures you see in your branch is because master wasn't tidy (this often happens when a new Perl::Tidy is released - it's very annoying).

**Comment by kenneth-olwing** [comment]
Right, thanks. Yes, there's some version mismatch, I don't think I can't make all test builds to pass tidy checks and whatnot, but I hope I've gotten the intended functionality correct.

Unless you immediately have something I've dropped on the floor, I think I'll leave it be for your perusal at this time, sound ok?

**Comment by autarch** [comment]
Hi Kenneth, I just released a 0.76-TRIAL version with this change included. Please let me know if it works for you.

I'm also going to post on my blog and ask people to test it. This is a big change and I'm a bit nervous that it could break some plugins or codebase out there, so I want to get some confirmation that it works in the wild.

**Comment by kenneth-olwing** [comment]
Completely understandable. Thanks for your work!

**Comment by kenneth-olwing** [comment]
Hi, back in the saddle after xmas/newyears. GreEtings!

I'm getting various test errors/warnings at the moment with the trial, so be sure to keep it on hold :-)

I've literally just tried various 'cpanm --test-only' on the tarball so I have no better insight yet. However, on both Linux/Windows I get warnings like:

t/Basic.t ...................... 81/? could not find tidyall.ini or .tidyallrc upwards from '/tmp/Code-TidyAll-f582/global_ignore' at /home/nfs/oktk/.cpanm/work/1578046183.15903/Code-TidyAll-0.76/lib/Code/TidyAll.pm line 736.
could not find tidyall.ini or .tidyallrc upwards from '/tmp/Code-TidyAll-f582/plugin_ignore' at /home/nfs/oktk/.cpanm/work/1578046183.15903/Code-TidyAll-0.76/lib/Code/TidyAll.pm line 736.

(on windows also test 104, same thing).

On Windows it appears to give me different things depending on what else I have installed, although this is the simplest:

t/Plugin-GenericTransformer.t .. 1/?
#   Failed test 'new contents [generic transformer preserves linefeed line endings]'
#   at t/lib/TestFor/Code/TidyAll/Plugin.pm line 70.
#   (in TestFor::Code::TidyAll::Plugin::GenericTransformer->test_main)
# +---+------------------------------+---------------------+
# | Ln|Got                           |Expected             |
# +---+------------------------------+---------------------+
# *  1|'61.0D.0A.62.0D.0A.63.0D.0A'  |'61.0A.62.0A.63.0A'  *
# +---+------------------------------+---------------------+

On another setup, same base Strawberry Perl but with more modules installed I get more confusing errors, like:

#   Failed test 'new contents [generic transformer preserves linefeed line endings]'
#   at t/lib/TestFor/Code/TidyAll/Plugin.pm line 70.
#   (in TestFor::Code::TidyAll::Plugin::GenericTransformer->test_main)
# +---+--------+---+-----------------------+
# | Ln|Got     | Ln|Expected               |
# +---+--------+---+-----------------------+
# *  1|'a\r\n  *  1|'61.0A.62.0A.63.0A'\n  *
# *  2|b       *   |                       |
# *  3|c       *   |                       |
# *  4|'       *   |                       |
# +---+--------+---+-----------------------+

which makes no sense to me. Plus also more errors similar to this. Shouldn't be like that of course so I'm going to try to track it down...

For reference, I'm currently using Perl 5.30.0 at this point. On Linux, it's home built from source, on Windows it's strawberry-perl-5.30.0.1-64bit-portable.zip from strawberry.org.
Basically, it's: 
1) Unzip
2) Open a cmd.exe and run the '<unzippedroot>\portableshell.bat'
3) cpanm Code::TidyAll (to install the 0.75 and it's dependencies)
4) cpanm -v --test-only <fullpathto076trial>

**Comment by kenneth-olwing** [comment]
Whoa, the cut and pasted lines got a bit bigger than I expected...hope you can read it anyway...:-)

**Comment by kenneth-olwing** [comment]
FYI, the "t/Basic.t ...................... 81/? could not find tidyall.ini or .tidyallrc upwards from ..." warning bit appears to have been in the 0.75 dist as well so it's just a minor nuisance in drawing eyes to it...

**Comment by kenneth-olwing** [comment]
Tiny error in t\lib\TestFor\Code\TidyAll\Plugin\GenericTransformer.pm, line 44:
    $lf_file->spew($lf);
should be
    $lf_file->spew_raw($lf);
This fixes the test error.

Next, I'll track down why I get more and, and more nonsensical, errors when I have a more fully configured Perl.

**Comment by autarch** [comment]
Yep, I'd found that missing `spew_raw` while working on another branch off this one.

**Comment by autarch** [comment]
The warning from `t/Basic.t` is expected and not a new issue (though it'd be nice to shut it up).

**Comment by kenneth-olwing** [comment]
The issue turned out to be a sensitivity to one of my Perls installed in a path with spaces (eg. 'C:\Program Files\Perl\....'.

I've pushed a branch 'ken1-pr97-fixes' to my fork to merge from. It fixes three things:
1) A general error testing file contents
2) Several errors caused by Perl in path with spaces
3) Removes nuisance output regarding missing tidyall.ini/.tidyallrc

Not sure how you like to do things - a new trial 0.76 or 0.77? Either way, please put up a new one...:-)

Thanks,
ken1

**Comment by autarch** [comment]
I just made a 0.77 trial release.

**Comment by kenneth-olwing** [comment]
Great - my initial scenario that prompted this PR seems to be fixed.

On the minus side so to speak, I don't really use any particular plugins beyond tidy/critic so more input from others if possible is certainly a good idea...

Thanks for your help!
